### PR TITLE
EARTH-1474: Set up new Earth Matters Footer

### PR DESCRIPTION
**Pull request by jdwjdwjdw**
# READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- TL;DR - Setting up new Earth Matters Footer

# Needed By (Date)
- When does this need to be merged by?

# Steps to Test

1. Do this
2. Then this
3. Then this

# Affected versions or modules
- Does this PR impact any particular module, version, or pull request?

# Associated Issues and/or People
- [EARTH-1474](https://stanfordits.atlassian.net/browse/EARTH-1474): Set up new Earth Matters Footer
- [Gitpod snapshot](https://gitpod.io#snapshot/c271483d-9fa3-45a5-a793-2f6fdbd85701)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: modules/stanford_earth_contact_footer/src/Plugin/Block/EarthContactFooterBlock.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+    $this->configuration['stanford_earth_contact_footer_settings'] = $form_state->getValue('stanford_earth_contact_footer_block_settings');
+  }
```
Comment (pookmish) [inline modules/stanford_earth_contact_footer/src/Plugin/Block/EarthContactFooterBlock.php]: You really don't need any of this since it doesn't do anything. You can delete these lines

File: modules/stanford_earth_contact_footer/stanford_earth_contact_footer.module [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * Implements hook_theme().
+ */
+
+function stanford_earth_contact_footer_theme() {
+  return array(
+   'stanford_earth_contact_footer_block' => array(
+            'variables' => NULL,
+            'template' => 'block--stanford_earth_contact_footer',
+        ),
+  );
```
Comment (pookmish) [inline modules/stanford_earth_contact_footer/stanford_earth_contact_footer.module]: Some code standard fixes
```suggestion
  return [
    'stanford_earth_contact_footer_block' => [
      'variables' => NULL,
      'template' => 'block--stanford_earth_contact_footer',
    ],
  ];
```

File: modules/stanford_earth_contact_footer/stanford_earth_contact_footer.module [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+/**
+ * Implements hook_theme().
+ */
+
```
Comment (pookmish) [inline modules/stanford_earth_contact_footer/stanford_earth_contact_footer.module]: Remove the empty line so the docblock above is directly above the function

File: modules/stanford_earth_matters_footer/src/Plugin/Block/EarthMattersFooterBlock.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+    $this->configuration['stanford_earth_matters_footer_settings'] = $form_state->getValue('stanford_earth_matters_footer_block_settings');
+  }
```
Comment (pookmish) [inline modules/stanford_earth_matters_footer/src/Plugin/Block/EarthMattersFooterBlock.php]: Delete these like the other one.

File: modules/stanford_earth_matters_footer/stanford_earth_matters_footer.module [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+ * Implements hook_theme().
+ */
+
+function stanford_earth_matters_footer_theme() {
+  return array(
+   'stanford_earth_matters_footer_block' => array(
+            'variables' => NULL,
+            'template' => 'block--stanford_earth_matters_footer',
+        ),
+  );
```
Comment (pookmish) [inline modules/stanford_earth_matters_footer/stanford_earth_matters_footer.module]:
```suggestion
 */
function stanford_earth_matters_footer_theme() {
  return [
    'stanford_earth_matters_footer_block' => [
      'variables' => NULL,
      'template' => 'block--stanford_earth_matters_footer',
    ],
  ];
```

File: modules/stanford_earth_contact_footer/src/Plugin/Block/EarthContactFooterBlock.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Drupal\stanford_earth_contact_footer\Plugin\Block;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\term_condition\Plugin\Condition\Term;
```
Comment (pookmish) [inline modules/stanford_earth_contact_footer/src/Plugin/Block/EarthContactFooterBlock.php]: I believe the only one of these that is actually needed is the `BlockBase`. All others can be deleted

File: modules/stanford_earth_matters_footer/src/Plugin/Block/EarthMattersFooterBlock.php [hunk 7]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Drupal\stanford_earth_matters_footer\Plugin\Block;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\term_condition\Plugin\Condition\Term;
```
Comment (pookmish) [inline modules/stanford_earth_matters_footer/src/Plugin/Block/EarthMattersFooterBlock.php]: I believe the only one of these that is actually needed is the `BlockBase`. All others can be deleted

File: modules/stanford_earth_contact_footer/templates/block--stanford_earth_contact_footer.html.twig [hunk 8]
```diff
@@ -0,0 +1,80 @@
+{#
```
Comment (pookmish) [inline modules/stanford_earth_contact_footer/templates/block--stanford_earth_contact_footer.html.twig]: normally with template files we keep them as hyphens instead of underscores. Lets adjust those here and in the `hook_theme()` functions

**Review by pookmish — CHANGES_REQUESTED** [review]

**Comment by jdwjdwjdw** [comment]
Thanks @pookmish! Updated and appears to be working as normal in my gitpod.

**Review by pookmish — APPROVED** [review]
Looks good. Great Job!
